src/transforms/basic.js

```diff
@@ -32,6 +34,20 @@ function composeTransform(t1, t2) {
   };
 }
 
+export function composeInitialize({initialize: i1, ...options} = {}, i2) {
+  return i1 == null
+  ? {...options, initialize: i2}
+  : {
+    ...options,
+    initialize(data, facets, channels, scales, dimensions) {
+      let newChannels;
+      ({data, facets, channels: newChannels} = i1.call(this, data, facets, channels, scales, dimensions));
+      Object.assign(channels, newChannels);
```

Comment:
This is mutating the input argument _channels_. We should avoid mutating arguments.